engine/gamesys/src/gamesys/components/comp_sprite.cpp

```diff
@@ -1665,64 +1688,19 @@ namespace dmGameSystem
 
-    static void UpdateTransforms(SpriteWorld* sprite_world, bool sub_pixels)
+    static void UpdateTransforms(SpriteComponent* component, bool scale_along_z, bool sub_pixels)
     {
-        DM_PROFILE("UpdateTransforms");
-
-        dmArray<SpriteComponent>& components = sprite_world->m_Components.GetRawObjects();
-        uint32_t n = components.Size();
-
-        bool scale_along_z = false;
-        if (n > 0)
-        {
-            SpriteComponent* c = &components[0];
-            scale_along_z = dmGameObject::ScaleAlongZ(dmGameObject::GetCollection(c->m_Instance));
-        }
-        // Note: We update all sprites, even though they might be disabled, or not added to update
-
-        if (scale_along_z)
-        {
-            for (uint32_t i = 0; i < n; ++i)
-            {
-                SpriteComponent* c = &components[i];
-                Matrix4 local = dmTransform::ToMatrix4(dmTransform::Transform(c->m_Position, c->m_Rotation, 1.0f));
-                Matrix4 world = dmGameObject::GetWorldMatrix(c->m_Instance);
-                Vector3 size( c->m_Size.getX() * c->m_Scale.getX(), c->m_Size.getY() * c->m_Scale.getY(), 1);
-                c->m_World = dmVMath::AppendScale(world * local, size);
-                // we need to consider the full scale here
-                // I.e. we want the length of the diagonal C, where C = X + Y
-                float radius_sq = dmVMath::LengthSqr((c->m_World.getCol(0).getXYZ() + c->m_World.getCol(1).getXYZ()) * 0.5f);
-                sprite_world->m_BoundingVolumes[i] = radius_sq;
-            }
-        }
-        else
-        {
-            for (uint32_t i = 0; i < n; ++i)
-            {
-                SpriteComponent* c = &components[i];
-                Matrix4 local = dmTransform::ToMatrix4(dmTransform::Transform(c->m_Position, c->m_Rotation, 1.0f));
-                Matrix4 world = dmGameObject::GetWorldMatrix(c->m_Instance);
-                Matrix4 w = dmTransform::MulNoScaleZ(world, local);
-                Vector3 size( c->m_Size.getX() * c->m_Scale.getX(), c->m_Size.getY() * c->m_Scale.getY(), 1);
-                c->m_World = dmVMath::AppendScale(w, size);
-                // we need to consider the full scale here
-                // I.e. we want the length of the diagonal C, where C = X + Y
-                float radius_sq = dmVMath::LengthSqr((c->m_World.getCol(0).getXYZ() + c->m_World.getCol(1).getXYZ()) * 0.5f);
-                sprite_world->m_BoundingVolumes[i] = radius_sq;
-            }
-        }
-
-        // The "sub_pixels" is set by default
+        Matrix4 local = dmTransform::ToMatrix4(dmTransform::Transform(component->m_Position, component->m_Rotation, 1.0f));
+        Matrix4 world = dmGameObject::GetWorldMatrix(component->m_Instance);
+        Vector3 size( component->m_Size.getX() * component->m_Scale.getX(), component->m_Size.getY() * component->m_Scale.getY(), 1);
+        Matrix4 w = scale_along_z ? world * local : dmTransform::MulNoScaleZ(world, local);
```

Comment:
Collapse the same code into smaller section and use here a ternary operator. I believe that CPU should make branch prediction optimization here because `scale_along_z` and `sun_pixels` doesn't change during the frame.

Reply:
In any case it will be merge conflict with your changes (for example, I moved bounding box assignment out of the function). I suggest to discuss the order of merging. You will be the first then I will merge my changes (if changes will pass the review)